src/test/java/spoon/test/processing/ProcessingTest.java

```diff
@@ -41,6 +44,20 @@ public void testInsertBegin() throws Exception {
 							+ constructor.getSimpleName(), "int i = 0;",
 					constructor.getBody().getStatement(1).toString());
 		}
+			
+		CtConstructor<?> constructor = type.getConstructor(type.getFactory().Type().INTEGER_PRIMITIVE);
+    String myBeforeStatementAsString = "int before";
+    for (CtSwitch<?> ctSwitch : constructor.getElements(new TypeFilter<CtSwitch<?>>(CtSwitch.class))) {
+      ctSwitch.insertBefore(type.getFactory().Code()
+        .createCodeSnippetStatement(myBeforeStatementAsString));
+    }
+    assertEquals("insert has not been done at the right position", myBeforeStatementAsString, constructor.getBody().getStatement(3).toString());
+    assertEquals("insert has not been done at the right position", myBeforeStatementAsString, constructor.getBody().getStatement(5).toString());
```

Comment:
No, the statement 5 is not a break, because getStatements only returns the statements which have the method block as parent.